xds/internal/resolver/xds_resolver_test.go

```diff
@@ -104,12 +107,12 @@ type testClientConn struct {
 }
 
 func (t *testClientConn) UpdateState(s resolver.State) error {
-	t.stateCh.Send(s)
+	t.stateCh.Replace(s)
```

Comment:
FYI: this seems okay as long as everywhere we read from this channel is resilient - as in, if the expected value isn't read, it reads again.  Otherwise, you can run into races if you read from the channel before it's written.

And even then, we can have false positives.  E.g.:

1. Random setup thing happens that pushes the expected value.
2. Test condition occurs, which will result in pushing something bad here.
3. Read expected value from channel and pass test.
4. Bad value we were hoping to test is actually pushed.

Forcing pulling every update off the channel effectively ensures the above won't happen.

Reply:
It seems both options here have problems, so I'm fine with this change.